### PR TITLE
set default_server also for https

### PR DESCRIPTION
**Pull request by webner**
DEFAULT_HOST was not respected when using https

### CODE REVIEW

**Comment by md5** [comment]
:+1:

**Comment by jwilder** [comment]
This fails for me when running just one container with a `VIRTUAL_HOST` but not starting `nginx-proxy` with `-e DEFAULT_HOST=foo.com`.

```
dockergen.1 | 2015/07/25 22:50:01 template error: template: nginx.tmpl:75:48: executing "nginx.tmpl" at <eq $.Env.DEFAULT_HOS...>: error calling eq: invalid type for comparison
```

File: nginx.tmpl [hunk 1]
```diff
@@ -72,6 +72,9 @@ upstream {{ $host }} {
 {{ end }}
 }
 
+{{ $default_host := or ($.Env.DEFAULT_HOST) "" }}
+{{ $default_server := index (dict $host "" $default_host "default_server") $host }}
```
Comment (webner) [inline nginx.tmpl]: this is little bit hacky, but I didn't find another way to set a variable depending on another variable value.

Perhaps we could add a "when" function to docker-gen?
$default_server := when (eq $host $default_host) "default_server" ""

**Comment by Informatic** [comment]
Are there plans on merging that in near future?

**Comment by jwilder** [comment]
Thanks @webner !

**Pull request merged**